server/sublist.go

```diff
@@ -149,8 +150,24 @@ func (s *Sublist) RegisterNotification(subject string, notify chan<- bool) error
 		return ErrNilChan
 	}
 
+	var hasInterest bool
 	r := s.Match(subject)
-	hasInterest := len(r.psubs)+len(r.qsubs) > 0
+
+	if len(r.psubs)+len(r.qsubs) > 0 {
+		for _, sub := range r.psubs {
+			if string(sub.subject) == subject {
```

Comment:
not sure all the cases that would hit this code so just a question - this will prevent `>` from matching, good, but what about checking for `x.>` for a message on `x.y`, should we support those wildcards?

Reply:
Slippery slope, I think we need to only match explicit.